XenModel/Alerts/PolicyAlert.cs

```diff
@@ -43,16 +45,51 @@ public class PolicyAlert : IEquatable<PolicyAlert>
         public readonly string Type;
         public readonly string Text;
         public readonly DateTime Time;
+        public readonly string PolicyType;
+        public readonly int numberOfVMsFailed;
 
-        public PolicyAlert(long priority, string name, DateTime _time)
+        public PolicyAlert(long priority, string name, DateTime _time, string body, string policyName)
         {
             Type = (priority == 4 ? "info": "error");
             Time = _time;
-            Text = Message.FriendlyBody(name);
+            PolicyType = "VMSS";
+
+            if(Type == "info")
+            {
+                Text = Message.FriendlyBody(name);
+                return;
+            }
+
+            /* We reach here when message type is an error hence we need 
+             * to parse the message body accordingly */
+
+            numberOfVMsFailed = Regex.Matches(body, "VM:").Count;
+            string[] messageHeader = Regex.Split(body.Replace("\n", " "), "Details:");
+            Text = string.Format(Message.FriendlyBody(name), policyName, numberOfVMsFailed, "\n\n" + Message.FriendlyBody("vmss_details") + "\n");

```

Comment:
I think we should only add the "Details" part if we find detailed VM errors. 
So, we can remove the {2} parameter from Message.body-vmss_snapshot_failed, and add "Details:" further down and only if `vmList.Length > 1`. (Please note that I had also suggested to move "Details:" to Messages.resx)
Also, we can build the Text better with the help of a StringBuilder. Then there's no need for string concatenations and explicit addition of  the newline character, because we can do StringBuilder's methods (AppendLine, Append, AppendFormat..).